Add deken upload

ben-wes opened the pull request:
this PR was motivated by some users asking for a simple FluCoMa installation via deken.

* closes #69

it adds a new `deken-upload` workflow which is triggered after a successful release.

a `FluidCorpusManipulation` deken package is created and its name and content is printed in the workflow summary and uploaded as an artifact (see https://github.com/ben-wes/flucoma-pd/actions/runs/18846787301 as an example).

the actual upload to deken is only triggered if the release was made on the `production` branch. the upload requires `DEKEN_USERNAME` and `DEKEN_PASSWORD` secrets.

`README.deken.pd` is a new patch that automatically opens after the installation. it provides direct access to the preferences for adding path and library - and also a button for adding the flucoma folder to Pd's search path. unfortunately, this doesn't seem possible for the `fluid_libmanipulation` library, so it needs to be added by the user.

it also links to the object overview and displays the reference launcher: 

<img width="693" height="696" alt="image" src="https://github.com/user-attachments/assets/0e063cc1-b7db-4ee4-a93b-cf9b387b5ebd" />

feedback welcome - the workflow might not be the most elegant. and obviously, the patch is also mainly a suggestion (the use of the `add-to-path` message is slightly hacky, but should work well afaict).

... thank you for your work and for sharing FluCoMa with everyone!

tremblap commented:
... and I am still waiting for the puredata.org account creation for us to be able to upload.

ben-wes commented:
@jamesb93 this is now updated and should ideally work combined with https://github.com/flucoma/actions/pull/7 and a `v6` tag on the actions side.

i'm not sure it's acceptable that i now updated the `release.yml` to upload an additional archive for macos to simplify the extraction for the deken packaging? i probably missed some detail there in the new version - but i just couldn't get the `dmg` extraction to properly work again there.

for testing, we should now also get a deken release with a branch that is named `deken-*`. and the content of the deken package should also be logged for any other branch.

tremblap commented:
also I have received the login for flucoma - i can put the deets in a secret if @jamesb93 tells me how 😆

ben-wes commented:
squashed the changes to make the PR a bit cleaner.

tremblap commented:
ok @ben-wes and @jamesb93 what is needed for this to work now? I have a login, should we upload it in the secret and boom we're on?

jamesb93 commented:
I merged https://github.com/flucoma/actions/pull/7. 

I also spun up a PR for the v6:

https://github.com/flucoma/flucoma-pd/pull/114

Let's see!

jamesb93 commented:
Ah, I've noticed a problem here.

We only publish a nightly on `main` or `ci/*`, and deken only fires on `deken-*` or `main`. So we should merge, and then see what happens I think.

jamesb93 commented:
https://github.com/flucoma/flucoma-pd/actions/runs/20360991211
